promoter/database.go

```diff
@@ -162,30 +195,71 @@ OUTER:
 			time.Sleep(2 * time.Second) // sleep before retrying
 			continue OUTER              // try again
 		}
-		initialUpdates := make([]WatchedAddressUpdate, 0, len(toAdd)+len(toRemove))
+		toRemoveUpdates := make([]WatchedAddressUpdate, 0, len(toRemove))
+		toAddUpdates := make([]WatchedAddressUpdate, 0, len(toAdd))
+		unused := true
 		for _, addr := range toAdd {
-			initialUpdates = append(initialUpdates, WatchedAddressUpdate{
-				Address:       addr,
+			unused = unused && addr.Unused()
+			toAddUpdates = append(toAddUpdates, WatchedAddressUpdate{
+				Address:       addr.Address,
 				OperationType: operationTypeInsert,
 			})
 		}
 		for _, addr := range toRemove {
-			initialUpdates = append(initialUpdates, WatchedAddressUpdate{
+			toRemoveUpdates = append(toRemoveUpdates, WatchedAddressUpdate{
 				Address:       addr,
 				OperationType: operationTypeDelete,
 			})
 		}
-		updateFn(initialUpdates...)
 
-		// Start listening for future changes.
+		// To avoid resyncing skyd's wallet too often, we check all
+		// possible edge cases and make sure we only call updateFn with
+		// unused = false once.
+		if len(toAddUpdates) > 0 && len(toRemoveUpdates) == 0 {
+			err = updateFn(unused, toAddUpdates...)
+		} else if len(toAddUpdates) == 0 && len(toRemoveUpdates) > 0 {
+			err = updateFn(false, toRemoveUpdates...)
```

Comment:
This should be `true`, right?

Reply:
Not here. We don't know if the address has been used or not. Because it's missing from the db. So we assume it was used. Since this is usually only called on startup we accept the penalty of causing a sync.